djtriggers/models.py

```diff
@@ -136,11 +137,21 @@ def _handle_execution_failure(self, exception, use_statsd=False):
         :return: None
         """
         self.number_of_tries += 1
+        # Log message if starts retrying too much
         if self.number_of_tries > getattr(settings, 'DJTRIGGERS_TRIES_BEFORE_WARNING', 3):
+            # Set a limit for retries
+            if self.number_of_tries >= getattr(settings, 'DJTRIGGERS_TRIES_BEFORE_ERROR', 5):
+                # Set date_processed so it doesn't retry anymore
+                self.date_processed = timezone.now()
+                self.successful = False
```

Comment:
I have a second idea that I want to bounce off of you. Not sure whether it's better or worse.

- Implement nullable `date_failed`
- If the trigger was not processed successfully, fill in `date_failed` instead of `date_processed`
- Update the way we query for new tasks so that failed tasks aren't retried

I'm leaning towards your implementation.

Reply:
I considered that too, but ended up with the current.